pkgs/tools/security/gitleaks/default.nix

```diff
@@ -44,6 +45,8 @@ buildGoModule rec {
     command = "${pname} version";
   };
 
+  passthru.updateScript = nix-update-script { };
```

Comment:
Bot and `nix-update` worked fine in the past. Does adding `nix-update-script` solve an issue or is it just for convenience?

Reply:
It let me run `nix-shell maintainers/scripts/update.nix --argstr package gitleaks` to generate this PR, which did not work before adding this.

This is the current advice in `pkgs/README.md`; if the situation has changed, please open a PR for that file: https://github.com/NixOS/nixpkgs/blob/7e9f7ce40331de889033e2e82b2e1cd96f7cdabe/pkgs/README.md#automatic-package-updates

Reply:
Manually updating gitleaks with `nix-update` adds the changelog and the diff to the commit message which also forms the content for the PR description. 

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#writing-good-commit-messages
> Package version upgrades usually allow for simpler commit messages, including attribute name, old and new version, as well as a reference to the relevant release notes/changelog.

`nix-shell maintainers/scripts/update.nix --argstr package gitleaks` doesn't do [this]. I highly doubt that people will start using `update.nix` over `nix-update` when updating packages manually, if not necessary, as you have to create the commit message by yourself.

The README (https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#automatic-package-updates) also says:

> While not strictly necessary if the project is listed on [Repology](https://repology.org/)...

`passthru.updateScript = nix-update-script { };` would then only be for convenience.

Reply:
> Manually updating gitleaks with `nix-update` adds the changelog and the diff to the commit message which also forms the content for the PR description.

Thanks, I didn't know this! This is the first I'm hearing that we should run `nix-update` directly. Could you point me to the documentation that explains how to use it? All I see [in the README](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#how-are-update-scripts-executed) is this note, which explicitly says to run `update.nix`:

> Update scripts are to be invoked by the [automatic package update script](https://github.com/NixOS/nixpkgs/blob/master/maintainers/scripts/update.nix). You can run `nix-shell maintainers/scripts/update.nix` in the root of Nixpkgs repository for information on how to use it.

`nix-update` is mentioned in passing ("A common pattern is to use the [`nix-update-script`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ni/nix-update/nix-update-script.nix) attribute provided in Nixpkgs, which runs [`nix-update`](https://github.com/Mic92/nix-update)"), but it doesn't explain how to run `nix-update` yourself or recommend doing so.

I'm happy to run `nix-update` directly if that's the new recommendation, but I can't find any documentation to corroborate this.

Reply:
I'm fine with this line being here purely for convenience. I encourage you to open a PR updating the `pkgs/README.md` documentation if you'd like to steer people towards `nix-update`.